homeassistant/components/russound_rio/config_flow.py

```diff
@@ -0,0 +1,127 @@
+        """Handle a flow initialized by the user."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            name = user_input[CONF_NAME]
+            port = user_input[CONF_PORT]
+
+            try:
+                russ = Russound(self.hass.loop, host, port)
+                async with asyncio.timeout(5):
+                    await russ.connect()
+                    controllers = await russ.enumerate_controllers()
+                    metadata = find_primary_controller_metadata(controllers)
+                    if metadata:
+                        await self.async_set_unique_id(metadata[0])
+                    else:
+                        raise NoPrimaryControllerException
+                    await russ.close()
+                    self._abort_if_unique_id_configured(
+                        updates={CONF_HOST: host, CONF_PORT: port}
+                    )
+            except RUSSOUND_RIO_EXCEPTIONS as err:
+                _LOGGER.error("Could not connect to Russound RIO: %s", err)
+                errors["base"] = "cannot_connect"
+            except NoPrimaryControllerException as err:
+                _LOGGER.error(
+                    "Russound RIO device doesn't have a primary controller: %s", err
+                )
+                errors["base"] = "no_primary_controller"
+            else:
+                data = {
+                    CONF_HOST: host,
+                    CONF_NAME: name,
+                    CONF_PORT: port,
+                }
+
+                return self.async_create_entry(title=name, data=data)
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_import(
+        self, import_config: dict[str, Any]
+    ) -> ConfigFlowResult:
+        """Attempt to import the existing configuration."""
+        self._async_abort_entries_match({CONF_HOST: import_config[CONF_HOST]})
+        host = import_config[CONF_HOST]
+        name = import_config[CONF_NAME]
+        port = import_config.get(CONF_PORT, 9621)
+
+        # Connection logic is repeated here since this method will be removed in future releases
+        try:
+            russ = Russound(self.hass.loop, host, port)
+            async with asyncio.timeout(5):
+                await russ.connect()
+                controllers = await russ.enumerate_controllers()
+                metadata = find_primary_controller_metadata(controllers)
+                if metadata:
+                    await self.async_set_unique_id(metadata[0])
+                else:
+                    return self.async_abort(
+                        reason="no_primary_controller", description_placeholders={}
+                    )
+                await russ.close()
+                self._abort_if_unique_id_configured(
+                    updates={CONF_HOST: host, CONF_PORT: port}
+                )
```

Comment:
Let's not do this, because if I still have the YAML and I deleted the imported config entry, and created a new one, because I got a new IP address (its a bit tricky, but it explains it well), it would overwrite that entry with the YAML (old) info